malloc.c

```diff
@@ -239,7 +239,7 @@
 #define PERL_IN_MALLOC_C
 #include "perl.h"
 #if defined(MULTIPLICITY)
-#    define croak2	Perl_croak_nocontext
+#    define croak2	croak
```

Comment:
I missed this before. You can't use `croak` here because `croak2` (and `emergency_sbrk_croak` further down) are called from functions without a `pTHX` parameter.